### PR TITLE
fix(web): Fix how layer is separated from key name

### PR DESCRIPTION
**Pull request by darcywong00**
Investigation of #3569 

The touch layout keyboard accesses `ctrl-alt` and `shift-ctrl-alt` layers on some longpresses (e.g. "numeric" layer <kbd>[</kbd> key). KMW was just doing a `split('-') to separate the layer from the key name.

This fixes KMW to account for those layers.

### CODE REVIEW

**Comment by darcywong00** [comment]
I intend to keep this as a draft PR (contains engineering code) primarily to discuss the KMW code block before finalizing to a separate branch.

File: web/source/kmwembedded.ts [hunk 1]
```diff
@@ -386,14 +386,19 @@ namespace com.keyman.text {
...
-      keyName=keyName.replace('popup-',''); //remove popup prefix if present (unlikely)      
-      
-      var t=keyName.split('-'),layer=(t.length>1?t[0]:core.keyboardProcessor.layerId);
-      keyName=t[t.length-1];
+      keyName=keyName.replace('popup-',''); //remove popup prefix if present (unlikely)
+      // Can't just split on '-' because some layers like ctrl-shift contain it.
+      let separatorIndex = keyName.lastIndexOf('-');
+      var layer = core.keyboardProcessor.layerId;
+      if (separatorIndex > 0) {
+        layer = keyName.substring(0, separatorIndex);
```
Comment (darcywong00) [inline web/source/kmwembedded.ts:394]: I originally had
`layer = keyName.substring(0, separatorIndex-1)`

but that was giving me layer `ctrl-al` 😕

Comment (jahorton) [reply web/source/kmwembedded.ts:394]: What you now have is right.  The index of a character in a string matches the length of the substring before it, after all.

**Review by jahorton — COMMENTED** [review]
Seems good to me so far, though I'd want to test things more thoroughly to be sure.  Let me know if you want that extra testing sooner rather than later.

**Comment by darcywong00** [comment]
> Seems good to me so far, though I'd want to test things more thoroughly to be sure. Let me know if you want that extra testing sooner rather than later.

k, I'll move the changes to a separate branch / PR for more rigorous testing.

**Pull request closed without merging**